src/replicator.rs

```diff
@@ -627,6 +633,89 @@ impl Replicator {
             CBLReplicator_SetSuspended(self.get_ref(), suspended);
         }
     }
+
+    /** Returns the replicator's current status. */
+    pub fn status(&self) -> ReplicatorStatus {
+        unsafe { CBLReplicator_Status(self.get_ref()).into() }
+    }
+
+    /** Indicates which documents have local changes that have not yet been pushed to the server
+    by this replicator. This is of course a snapshot, that will go out of date as the replicator
+    makes progress and/or documents are saved locally. */
+    pub fn pending_document_ids(&self) -> Result<HashSet<String>> {
+        unsafe {
+            let mut error = CBLError::default();
+            let docs: FLDict =
+                CBLReplicator_PendingDocumentIDs(self.get_ref(), std::ptr::addr_of_mut!(error));
+
+            check_error(&error).and_then(|()| {
+                if docs.is_null() {
+                    return Err(Error::default());
+                }
+
+                let dict = Dict::wrap(docs, self);
+                Ok(dict.to_keys_hash_set())
+            })
+        }
+    }
+
+    /** Indicates whether the document with the given ID has local changes that have not yet been
+    pushed to the server by this replicator.
+
+    This is equivalent to, but faster than, calling \ref pending_document_ids and
+    checking whether the result contains \p docID. See that function's documentation for details. */
+    pub fn is_document_pending(&self, doc_id: &str) -> Result<bool> {
+        unsafe {
+            let mut error = CBLError::default();
+            let result = CBLReplicator_IsDocumentPending(
+                self.get_ref(),
+                from_str(doc_id).get_ref(),
+                std::ptr::addr_of_mut!(error),
+            );
+            check_error(&error).map(|_| result)
+        }
+    }
+
+    /**
+     Adds a listener that will be called when the replicator's status changes.
+    */
+    #[must_use]
+    pub fn add_change_listener(mut self, listener: ReplicatorChangeListener) -> Self {
+        let listener = unsafe {
+            let listener = Box::new(listener);
+            let ptr = Box::into_raw(listener);
+            Listener::new(
+                ListenerToken::new(CBLReplicator_AddChangeListener(
+                    self.get_ref(),
+                    Some(c_replicator_change_listener),
+                    ptr.cast(),
+                )),
+                Box::from_raw(ptr),
+            )
+        };
+        self.change_listener.push(listener);
+        self
+    }
+
+    /** Adds a listener that will be called when documents are replicated. */
+    #[must_use]
```

Comment:
Should it be must_use ?

Reply:
Yes

Reply:
If the `Replicator` structure is not used then its lifetime necessarily stops at the end of the scope, so the replication stops